src/code/z_olib.c

```diff
@@ -248,45 +248,49 @@ Vec3s* OLib_Vec3fDiffBinAng(Vec3s* dest, Vec3f* a, Vec3f* b) {
 }
 
 /**
- * Gets a x,y,z position diff depending on the dbCamera mode
+ * Gets a x,y,z position diff depending on the mode
  */
-void OLib_DbCameraVec3fDiff(PosRot* a, Vec3f* b, Vec3f* dest, s16 mode) {
-    VecSph sph;
+void OLib_Vec3fDiff(PosRot* a, Vec3f* b, Vec3f* dest, s16 mode) {
+    VecGeo geo;
 
     switch (mode) {
-        case 1:
-            OLib_Vec3fDiffToVecSphGeo(&sph, &a->pos, b);
-            sph.yaw -= a->rot.y;
-            OLib_VecSphGeoToVec3f(dest, &sph);
+        case OLIB_ADD_OFFSET:
+            OLib_Vec3fDiffToVecGeo(&geo, &a->pos, b);
+            geo.yaw -= a->rot.y;
+            OLib_VecGeoToVec3f(dest, &geo);
             break;
-        case 2:
+
+        case OLIB_ADD:
             dest->x = b->x - a->pos.x;
             dest->y = b->y - a->pos.y;
             dest->z = b->z - a->pos.z;
             break;
-        default:
+
+        default: // OLIB_COPY
             *dest = *b;
             break;
     }
 }
 
 /**
- * Gets a x,y,z position sum depending on the dbCamera mode
+ * Gets a x,y,z position sum depending on the mode
  */
-void OLib_DbCameraVec3fSum(PosRot* a, Vec3f* b, Vec3f* dest, s16 mode) {
-    VecSph sph;
+void OLib_Vec3fAdd(PosRot* a, Vec3f* b, Vec3f* dest, s16 mode) {
+    VecGeo geo;
 
     switch (mode) {
         case 1:
```

Comment:
Should this be using the `OLIB_ADD_OFFSET` constant?

Reply:
Yeah you're right, this enum should be used here